src/aleph/services/cost.py

```diff
@@ -77,7 +78,18 @@ def get_additional_storage_price(
 def compute_cost(session: DbSession, content: ExecutableContent) -> Decimal:
     is_microvm = isinstance(content, ProgramContent) and not content.on.persistent
     compute_unit_cost: Decimal = Decimal("200.0") if is_microvm else Decimal("2000.0")
+    cpu: Decimal = Decimal(content.resources.vcpus)
+    memory: Decimal = Decimal(math.ceil(content.resources.memory / 2000))
+    compute_unit_multiplier = cpu if cpu >= memory else memory
+    environment_internet: Decimal = Decimal(0)
+
+    if isinstance(content, ProgramContent) and content.environment.internet:
+        environment_internet = Decimal(2)
 
-    return (compute_unit_cost * content.resources.vcpus) + get_additional_storage_price(
+    compute_units_required = compute_unit_multiplier
+    compute_unit_multiplier = environment_internet or 1
```

Comment:
As mentioned in the comment above, this can be simplified to

```
compute_unit_multiplier = 1
if content.environment.internet:
  compute_unit_multiplier += 1
```